File: src/main/java/com/fish/entity/pojo/ColumnTopic.java
package com.fish.entity.pojo;


import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.Table;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import java.io.Serializable;

/**
 *  实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@NoArgsConstructor
@AllArgsConstructor
@Table(value = "oj_column_topic")
public class ColumnTopic implements Serializable {
    @Id
    @NotNull(message = "栏目id不能为空")
    private Long columnId;
    @Id
    @NotNull(message = "题目id不能为空")
    private Long topicId;

    public Long getColumnId() {
        return columnId;
    }

    public void setColumnId(Long columnId) {
        this.columnId = columnId;
    }

    public Long getTopicId() {
        return topicId;
    }

    public void setTopicId(Long topicId) {
        this.topicId = topicId;
    }
}

File: src/main/kotlin/com/fish/entity/pojo/ColumnTopic.kt
package com.fish.entity.pojo

import com.mybatisflex.annotation.Id
import com.mybatisflex.annotation.Table
import jakarta.validation.constraints.NotNull
import java.io.Serializable

/**
 * 实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Table(value = "oj_column_topic")
open class ColumnTopic: Serializable {
    @Id
    @NotNull(message = "栏目id不能为空")
    var columnId: Long? = null
    @Id
    @NotNull(message = "题目id不能为空")
    var topicId: Long? = null
}
